lib/Echidna/Campaign.hs

```diff
@@ -167,11 +176,12 @@ campaign :: ( MonadCatch m, MonadRandom m, MonadReader x m, Has TestConf x, Has
          -> [SolTest]           -- ^ Tests to evaluate
          -> Maybe GenDict       -- ^ Optional generation dictionary
          -> m Campaign
-campaign u v w ts d = let d' = fromMaybe mempty d in fmap (fromMaybe mempty) (view (hasLens . to knownCoverage))
-  >>= \c -> execStateT runCampaign (Campaign ((,Open (-1)) <$> ts) c d') where
-    step        = runUpdate (updateTest v Nothing) >> u >> runCampaign
+campaign u v w ts d = let d' = fromMaybe mempty d in fmap (fromMaybe mempty) (view (hasLens . to knownCoverage)) >>= \c -> do
+  g <- view (hasLens . to (maybe (mkStdGen 0) mkStdGen . seed))
```

Comment:
I don't think we should just silently set the seed to zero here. Maybe this could take a `StdGen` instead?

Reply:
fixed